erpnext/accounts/party.py

```
@@ -685,22 +685,17 @@ def validate_due_date_with_template(posting_date, due_date, bill_date, template_
-			msgprint(
-				_("Note: Due Date exceeds allowed {0} credit days by {1} day(s)").format(
-					party_type, date_diff(due_date, default_due_date)
-				)
-			)
-		else:
-			frappe.throw(_("Due Date cannot be after {0}").format(formatdate(default_due_date)))
+	# Enforce strict equality: when a payment terms template is set,
+	# the document due_date must exactly match the calculated template date.
+	# If it does not, raise a ValidationError to prevent manual overrides.
+	if getdate(due_date) != getdate(default_due_date):
+		raise frappe.ValidationError("You cannot manually change Due Date. It is locked by Payment Terms")
```

Ruff (0.14.8)

Avoid specifying long messages outside the exception class

(TRY003)